ios/RCTOneSignal/RCTOneSignalEventEmitter.m

```diff
@@ -389,19 +389,25 @@ + (void)sendEventWithName:(NSString *)name withBody:(NSDictionary *)body {
  */
 RCT_EXPORT_METHOD(sendOutcome:(NSString *)name :(RCTResponseSenderBlock)callback) {
     [OneSignal sendOutcome:name onSuccess:^(OSOutcomeEvent *outcome){
-        callback(@[[outcome jsonRepresentation]]);
+        if (outcome) {
```

Comment:
nil outcome represents a failure, shouldn't we return a failure text o something?

Reply:
not currently. we are calling `success(nil)` from the native side so it is technically considered a success. The JS-side function currently takes one callback which expects an `OutcomeEvent` object as you can tell by the TS typings. 

There are two ways to handle callback errors in JS [as documented](https://reactnative.dev/docs/native-modules-ios#callbacks):

1. expect two arguments in the callback e.g: `(error, outcomeEvent)`
2. expect two separate callbacks e.g: `(outcomeEvent) => {}, (error) => {}`

out of the two options, I would prefer the second one. However, this would require a change to the typings and documentation. Also, from inspecting the [native code](https://github.com/OneSignal/OneSignal-iOS-SDK/blob/master/iOS_SDK/OneSignalSDK/Source/OneSignalOutcomeEventsController.m) it seems that the call to `success(nil)` really only happens with unique outcomes as far as I can tell. So that means that we would only really need to add error handling in the bridge for `sendUniqueOutcome`. But adding it only to that function would not be consistent.

For the above reasons, I suggest we hold off adding any error handling or any changes to the bindings themselves until we add a `onFailure` block native side (TO DO). 

As a sort of compromise we can add native-side logging for the nil case but this won't be visible to JS developers unless they open up Xcode.

Reply:
I agree, there isn't a need to add to a signature for failures now if this isn't something both the OneSignal-Android-SDK and OneSignal-iOS-SDK can provide.